src/generators/hlopt.ml

```diff
@@ -1051,121 +1051,24 @@ let _optimize (f:fundecl) =
 
 let same_op op1 op2 =
 	match op1, op2 with
-	| OMov (a1,b1), OMov (a2, b2) -> a1 = a2 && b1 = b2
 	| OInt (r1,_), OInt (r2, _) -> r1 = r2
```

Comment:
Looking at this and other lines here, I don't understand why `OInt(1, 1)` and `OInt(1, 2)` should be considered the same op.

Reply:
It's `OInt reg * int index`, what only matters in optimized code is the used register / control flow, and not the index in global int table (e.g. in two different run, the same int 55 can have index 10 or 35). They are "fixed" by code related to c_remap_indexes.

I should double check if the replacement is always good, I'm trying to also remap field index but there are some errors x(

Reply:
`same_op` is confusing. I should probably rename the function but I don't have an good idea. Maybe `same_op_except_index`